service/controller/chart/v1/resource/release/resource.go

```diff
@@ -94,6 +100,31 @@ func (c *ReleaseState) IsEmpty() bool {
 	return c.Equals(ReleaseState{})
 }
 
+func isReleaseModified(a, b ReleaseState) bool {
+	// Version has changed so we need to update the Helm Release.
+	if a.Version != b.Version {
+		return true
+	}
+
+	// Values have changed so we need to update the values for the current
+	// Helm Release.
+	if !reflect.DeepEqual(a.Values, b.Values) {
+		return true
+	}
+
+	return false
+}
+
+func isReleaseInTransitionState(c ReleaseState) bool {
+	for _, status := range releaseTransitionStatuses {
```

Comment:
What if `releaseTransitionStatuses` would be a map (hash) instead of an array so we can do as below
```
if existed, _ := releaseTransitionStatuses[status]; existed {
return true
} else {
return false
}
```
it can reduce the computational cost.

Reply:
We usually do not have performance issues on this level. A map in this case would mean to maintain a key/value pair where we are only interested in one of two items of the pair. So I think the list is good enough.